src/main/java/athena/TaskList.java

```diff
@@ -156,10 +157,10 @@ public void editTask(int taskNumber, String name, String startTime, String durat
      * @param taskFilter The filter that decides which tasks are printed
      * @return Filtered task list
      */
-    public TaskList getFilteredList(TaskFilter taskFilter) {
+    public TaskList getFilteredList(TaskFilter taskFilter, ForecastFilter forecastFilter) {
         ArrayList<Task> filteredTasks = new ArrayList<>();
         for (Task task : tasks) {
-            if (taskFilter.isTaskIncluded(task)) {
+            if (taskFilter.isTaskIncluded(task) && forecastFilter.isTaskIncluded(task)) {
```

Comment:
I think we don't need to add another `ForecastFilter` parameter because `ForecastFilter` is a subclass of `TaskFilter`. In the `Timetable` class I elaborated on how this can be used.

With this, it will be possible to use any type of filters as we add more in the future (for example minus importance filter).